app/services/queue_submission.rb

```diff
@@ -6,18 +6,15 @@
 #
 class QueueSubmission < SimpleDelegator
   def call
-    success = ActiveRecord::Base.transaction do
+    update(state: Submission::QUEUED)
+
+    ActiveRecord::Base.transaction do
       begin
-        # double update otherwise `id` can be nil
-        update(state: Submission::QUEUED)
         update(job_id: DeliverSubmissionWorker.perform_async(id))
       rescue Redis::BaseConnectionError
+        update(state: Submission::UNQUEUED)
```

Comment:
It doesn't look like this update happens – possibly because it gets rolled back by the transaction.

**Steps to reproduce:**

1. Apply:

```diff
diff --git a/app/services/queue_submission.rb b/app/services/queue_submission.rb
index d16e0f0..ae8c3dc 100644
--- a/app/services/queue_submission.rb
+++ b/app/services/queue_submission.rb
@@ -10,7 +10,8 @@ class QueueSubmission < SimpleDelegator

     ActiveRecord::Base.transaction do
       begin
-        update(job_id: DeliverSubmissionWorker.perform_async(id))
+        #update(job_id: DeliverSubmissionWorker.perform_async(id))
+        raise Redis::BaseConnectionError
       rescue Redis::BaseConnectionError
         update(state: Submission::UNQUEUED)
         raise ActiveRecord::Rollback
```

2. Make a request
3. Inspect last submission state:

```ruby
Submission.last.state
```
  * **Expected:** 'unqueued'
  * **Actual:** 'queued'

Reply:
Oh of course, could've guess that by looking at the code. I got mistaken due to specs passing.